### ISSUE TITLE
No image preview in mounted location (fuse-zip)

### ISSUE THREAD

**Issue by maki-tompson** [ranger/ranger#1595]
#### Runtime Environment

 - Operating system and version: 
   * Xubuntu 18.04
 - Terminal emulator and version: 
   * rxvt-unicode (urxvt) v9.22 - released: 2016-01-23
   * xfce4-terminal 0.8.7.4 (Xfce 4.12)
 - Python version: 
   * python 2.7.15+
   * python 3.6.8
 - Ranger version/commit: 
   * ranger 1.8.1 (ubuntu 18.04.2 / python 2.7.15+)
   * ranger 1.9.2 (virtualenv / python 3.6.8)
 - Locale: 
   * `LANG=en_US.UTF-8`

#### Current Behavior

Ranger didn't show image preview in mounted zip archive (crash in debug mode).
Ranger can display image preview in extracted zip archive folder (archive and images ok).

#### Expected Behavior

Ranger should show image preview in mounted zip archive.
Ranger should show image preview in extracted zip archive folder.

#### Context

I wanna create ext which will mount zip files and allow navigate it via ranger. Image preview important.

Ranger configured properly and can display image preview.

```
## ~/.config/ranger/rc.conf
set preview_images true
#set preview_images_method urxvt-full

set draw_borders true
```

fuse-zip

```
Package: fuse-zip
Version: 0.4.4-1
Package: w3m
Version: 0.5.3-36build1
```

archive mounted properly and files accessible (thunar, feh, mc)

```
fuse-zip on /tmp/arch/mount type fuse.fuse-zip (rw,nosuid,nodev,relatime,user_id=1000,group_id=1000)
fuse-zip on /tmp/arch/sample type fuse.fuse-zip (rw,nosuid,nodev,relatime,user_id=1000,group_id=1000)
```

#### Steps to reproduce
1. Mount archive with some images and data (files/images/etc) in RO/RW mode `fuse-zip -r arch.zip /path/to/mount` or `fuse-zip arch.zip /path/to/mount`
2. Visit mount location `/path/to/mount` with ranger `ranger /path/to/mount`. 
3. There is no preview. (`ranger --debug /path/to/mount` => Crash)

#### Traceback

```
ranger version: 1.8.1, executed with python 2.7.15+
Locale: en_US.UTF-8
Current file: /tmp/arch/mount/01.jpg
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/ranger/core/main.py", line 145, in main
    fm.loop()
  File "/usr/lib/python2.7/dist-packages/ranger/core/fm.py", line 366, in loop
    ui.draw_images()
  File "/usr/lib/python2.7/dist-packages/ranger/gui/ui.py", line 355, in draw_images
    self.browser.columns[-1].draw_image()
  File "/usr/lib/python2.7/dist-packages/ranger/gui/widgets/pager.py", line 101, in draw_image
    self.fm.notify(e, bad=True)
  File "/usr/lib/python2.7/dist-packages/ranger/gui/widgets/pager.py", line 97, in draw_image
    self.wid, self.hei)
  File "/usr/lib/python2.7/dist-packages/ranger/ext/img_display.py", line 106, in draw
    start_y, width, height))
  File "/usr/lib/python2.7/dist-packages/ranger/ext/img_display.py", line 157, in _generate_w3m_input
    raise Exception('Failed to execute w3mimgdisplay', output)
Exception: ('Failed to execute w3mimgdisplay', [])
```
```
ranger version: ranger 1.9.2
Python version: 3.6.8 (default, Jan 14 2019, 11:02:34) [GCC 8.0.1 20180414 (experimental) [trunk revision 259383]]
Locale: en_US.UTF-8
Current file: '/tmp/arch/mount/01.jpg'

Traceback (most recent call last):
  File "/tmp/venv/lib/python3.6/site-packages/ranger/core/main.py", line 195, in main
    fm.loop()
  File "/tmp/venv/lib/python3.6/site-packages/ranger/core/fm.py", line 398, in loop
    ui.draw_images()
  File "/tmp/venv/lib/python3.6/site-packages/ranger/gui/ui.py", line 404, in draw_images
    self.browser.columns[-1].draw_image()
  File "/tmp/venv/lib/python3.6/site-packages/ranger/gui/widgets/pager.py", line 116, in draw_image
    self.fm.notify(ex, bad=True)
  File "/tmp/venv/lib/python3.6/site-packages/ranger/core/actions.py", line 172, in notify
    raise obj
  File "/tmp/venv/lib/python3.6/site-packages/ranger/gui/widgets/pager.py", line 111, in draw_image
    self.wid, self.hei)
  File "/tmp/venv/lib/python3.6/site-packages/ranger/ext/img_display.py", line 145, in draw
    input_gen = self._generate_w3m_input(path, start_x, start_y, width, height)
  File "/tmp/venv/lib/python3.6/site-packages/ranger/ext/img_display.py", line 209, in _generate_w3m_input
    raise ImageDisplayError('Failed to execute w3mimgdisplay', output)
ranger.ext.img_display.ImageDisplayError: ('Failed to execute w3mimgdisplay', [])
```

**Comment by nicholasdewaal** [comment]
After testing w3mdisplay alone in a fuse environment, it fails. It's a w3mdisplay problem, and not ranger.

In terminal, try running, but replacing your_image with an image you have in both a fuse and non-fuse environment:
    $FILENAME=your_image.jpg
    $width= your image's width
    $height= your image's height
    w3m_command="0;1;0;0;$width;$height;;;;;$FILENAME\n4;\n3;"
    W3MIMGDISPLAY="/usr/lib/w3m/w3mimgdisplay"
    echo -e $w3m_command | $W3MIMGDISPLAY

**Comment by toonn** [comment]
That's what the exception suggests, yes. Does this occur with other preview methods?

**Comment by disorder** [comment]
I've stumbled over this issue too, it is `w3mimgdisplay` issue.  Also it looks like some kind of feature (maybe some kind of sandbox since `w3m` is a web browser):

- it happens on FUSE mounts
- it doesn't happen if you mount with `-o allow_other`
- it happens in Debian package (built in debian branch)
- it doesn't happen in binary built from master branch of https://salsa.debian.org/debian/w3m.git

I don't actually use `ranger` but I found this issue.

TL;DR: there are at least two workarounds - build `w3mimgdisplay` from master branch or mount FUSE filesystems with `-o allow_other`.

**Comment by toonn** [comment]
@disorder, thanks for confirming this is an issue with `w3mimagedisplay`.
